core/src/Kokkos_View.hpp

```diff
@@ -494,8 +494,18 @@ constexpr bool is_assignable(const Kokkos::View<ViewTDst...>& dst,
 
 namespace Kokkos {
 
+// FIXME_OPENMPTARGET - The `declare target` is needed for the Intel GPUs with
+// the OpenMPTarget backend
+#if defined(KOKKOS_ENABLE_OPENMPTARGET) && defined(KOKKOS_ARCH_INTEL)
```

Comment:
That macro doesn't exist
```suggestion
#if defined(KOKKOS_ENABLE_OPENMPTARGET) && defined(KOKKOS_ARCH_INTEL_GPU)
```
but I really think it's more a compiler issue than an architecture issue.

Reply:
That should work with `KOKKOS_ENABLE_OPENMPTARGET`

Reply:
Or maybe `KOKKOS_ARCH_INTEL_GPU` ?

Reply:
lets do the compiler. But isn't it KOKKOS_COMPILER_INTEL_LLVM or something like that?

Reply:
I thought KOKKOS_COMPILER_INTEL is the classical one?

Reply:
```C++
#if defined(__INTEL_COMPILER)
#define KOKKOS_COMPILER_INTEL __INTEL_COMPILER
#elif defined(__INTEL_LLVM_COMPILER)
#define KOKKOS_COMPILER_INTEL __INTEL_LLVM_COMPILER
#elif defined(__ICC)
// Old define
#define KOKKOS_COMPILER_INTEL __ICC
#elif defined(__ECC)
// Very old define
#define KOKKOS_COMPILER_INTEL __ECC
#endif
```
is all we have.

Reply:
oh ok

Reply:
Yeah so that is fine then KOKKOS_COMPILER_INTEL

Reply:
Ok edited the code to protect the `declare target` directives with `KOKKOS_ENABLE_OPENMPTARGET` and `KOKKOS_COMPILER_INTEL`